src/remote_action_middleware.js

```diff
@@ -1,6 +1,15 @@
+import constants from '../src/constants';
+
 export default (socket) => (store) => (next) => (action) => {
   if (action.meta && action.meta.remote) {
     socket.emit(action.type, action);
   }
+
+  if(action.onSuccess) {
+    const { type, onSuccess } = action.onSuccess;
+
+    socket.on(type, onSuccess);
```

Comment:
Can you explain what this line is doing? Why do we need to add a listener to the socket here?